op-dispute-mon/mon/forecast.go

```diff
@@ -111,6 +115,20 @@ func (f *Forecast) forecastGame(game *monTypes.EnrichedGameData, metrics *foreca
 		return nil
 	}
 
+	// Games that have their block number challenged are won
+	// by the challenger since the counter is proven on-chain.
+	if game.BlockNumberChallenged {
+		f.logger.Debug("Found game with challenged block number",
+			"game", game.Proxy, "blockNum", game.L2BlockNumber, "agreement", agreement)
+		if !agreement {
+			metrics.AgreeChallengerAhead++
```

Comment:
This is an unexpected case and should log a warning.  I suspect we should just be setting `forecastStatus` appropriately (and skipping the `Resolve` call) then going through the same if block for updating metrics so that we get all the right warnings etc and don't duplicate logic.

Reply:
btw this also means we don't need any new alerts. If someone manages to challenge an l2 block incorrectly, we'll get a forecasted incorrect result alert.

Reply:
ah I see we were incrementing both - unfortunately that means the total number of games is reported incorrectly since you should be able to sum up the games in all categories to get the total, but we'd now be counting them twice.  If we want a metric counting the l2 block challenges, we'd need to make it a totally separate metric. It's not required for alerting but is kind of useful to see it easily.

Reply:
Right exactly. I would like to have a very simple, obvious metric so it is incredibly clear when the l2 block number is challenged. I do like the idea of setting `forecastStatus` though, will update that.